std/experimental/normint.d

```diff
@@ -0,0 +1,571 @@
+// Written in the D programming language.
+
+/**
+This module implements support for normalized integers.
+
+Authors:    Manu Evans
+Copyright:  Copyright (c) 2015, Manu Evans.
+License:    $(WEB boost.org/LICENSE_1_0.txt, Boost License 1.0)
+Source:     $(PHOBOSSRC std/experimental/normint.d)
+*/
+module std.experimental.normint;
+
+import std.traits : isIntegral, isSigned, isUnsigned, isFloatingPoint;
+static import std.algorithm;
+
+import std.experimental.color : FloatTypeFor;
+
+@safe pure nothrow @nogc:
+
+
+/**
+Normalized integers express a fractional range of values within an integer data type.
+
+Unsigned integers map the values 0 -> I.max to the fractional values 0.0 -> 1.0 in equal increments.
+Signed integers represent the values -I.max -> 0 -> I.max to fractional values -1.0 -> 0.0 -> 1.0 in equal increments. I.min is outside the nominal integer range and is clamped to represent -1.0.
+
```

Comment:
Maybe use the math interval `[a,b]` notation. IIRC it's already used in std.algorithm.